api/v1alpha1/bundle_types.go

```diff
@@ -51,13 +51,27 @@ type BundleSource struct {
 	Type string `json:"type"`
 	// Image is the bundle image that backs the content of this bundle.
 	Image *ImageSource `json:"image,omitempty"`
+	// Git is the git repository that backs the content of this Bundle.
+	Git *GitSource `json:"git,omitempty"`
 }
 
 type ImageSource struct {
 	// Ref contains the reference to a container image containing Bundle contents.
 	Ref string `json:"ref"`
 }
 
+type GitSource struct {
```

Comment:
Follow-up: add the comments + kubebuilder markers for these fields?

Reply:
Big +1, really good GoDoc is key on these APIs because it drives the content of `kubectl explain`

Reply:
For `Ref`, we should definitely include a statement in the GoDoc about what's valid. For example:
```go
// Ref configures the git source to clone a specific branch, tag, or commit 
// from the specified repo. Ref is required, and exactly one field within Ref
// is required. Setting more than one field or zero fields will result in an
// error
```

Reply:
I added some comments for each of the fields in the newly introduced git structs. They should be much more clear now. 

Regarding kubebuilder markers, I'm not sure which markers to add -- I know we wanted to enforce some union semantics but I don't think kubebuilder support those (since controller-gen does not). We have to manually modify the Bundle CRD manifest to get that functionality.